lms/djangoapps/experiments/permissions.py

```diff
@@ -29,3 +29,13 @@ def has_permission(self, request, view):
 class IsStaffOrReadOnly(BasePermission):
     def has_permission(self, request, view):
         return request.user.is_staff or request.method in SAFE_METHODS
+
+
+class IsStaffOrReadOnlyForSelf(BasePermission):
+    """
+    Grants access to staff or to user reading info about their own user
```

Comment:
Why do we need staff access if Optimizely makes the request on behalf of the user and any user can access their own data via the new endpoint, including staff users?

Reply:
Optimizely itself won't need this, but my thinking is:

(b) this is helpful for testing and troubleshooting if we want to call directly to look at results for different users